### PR TITLE
adding auto complete feature

### LINKED ISSUE #6: Autocomplete for cities
At the moment, when you type in a city, it won't help you finish your entry with suggestions. A cool feature would be if it had a drop down menu that suggested locations similar to the name. For example, if I type in `Cambridge`, it could tell me `Cambridge, MA, USA` or `Cambridge, UK` to help users get the right answers!

### PR DESCRIPTION
**Pull request by HetviSoni**
closes #6 

https://user-images.githubusercontent.com/76479104/180285006-dd5b04c5-9818-4e7f-8037-de2cbb616237.mp4

### CODE REVIEW

**Comment by sahibardj** [comment]
image


Auto complete feature

**Comment by sahibardj** [comment]
Deploy not working as the API key has not been added to the code.

File: src/Search.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import React, { Component } from "react";
+import { ReactSearchAutocomplete } from "react-search-autocomplete";
+import "./search.css";
+
+const autocompleteURL =
+  "https://autocomplete.search.hereapi.com/v1/autocomplete?";
+
+function autoCompleteCity(city) {
+  if (!city) return Promise.resolve("");
```
Comment (Shofiya2003) [inline src/Search.js:9]: Why are you returning empty string when city parameter is undefined. Why not an empty array?

File: src/Search.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+  const _handleOnSearch = async (searchText) => {
+    const items = await autoCompleteCity(searchText);
+    setItems(items);
+    setSearchText(searchText);
+  };
+
+  const _handleOnSelect = ({ name }) => {
+    setSearchText(name);
+    setCity(name);
```
Comment (Shofiya2003) [inline src/Search.js:48]: image
For some cities selected from the list, the weather API gives 404 not found error. Could you please just add a check that if the city gives the code 404 just display a message 'weather not found' or something.

**Review by Shofiya2003 — CHANGES_REQUESTED** [review]

File: src/Search.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+          <ReactSearchAutocomplete
+            items={items}
+            showNoResults={false}
+            onSearch={_handleOnSearch}
+            onSelect={_handleOnSelect}
+            autoFocus
+            styling={{
+              borderRadius: "12px",
+            }}
+            useCaching
```
[the remark this answers is not in the document]
Comment (sahibardj) [reply src/Search.js:64]: https://restfulapi.net/caching/

File: src/Search.js (hunk 2 above)
Comment (sahibardj) [reply src/Search.js:48]: I guess that is to be added in the weather part?

File: src/Search.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    setCity(name);
+  };
+
+  return (
+    <>
+      <div className="search-box">
+        <header className="box-header">
+          <ReactSearchAutocomplete
+            items={items}
+            showNoResults={false}
```
[the remark this answers is not in the document]
Comment (sahibardj) [reply src/Search.js:57]: Resolved

File: src/Search.js (hunk 1 above)
Comment (sahibardj) [reply src/Search.js:9]: resolved

File: src/Search.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (Shofiya2003) [reply src/Search.js:64]: how did you know that using useCaching prop would cache the API response passed to ReactSearchAutocomplete?

File: src/Search.js (hunk 2 above)
Comment (Shofiya2003) [reply src/Search.js:48]: Yup maybe we can create a new issue to handle that.

**Pull request closed without merging**